### PR TITLE
[fix](group commit) Fix compatibility issues on serializing and deserializing wal file

### PR DESCRIPTION
**Pull request by hust-hhb**
Now wal (be_exec_version is 3) deserializing to block using the newest way，after merging this commit https://github.com/apache/doris/commit/954750b59a8c6d6eb0525c5daa6ed26d5351b839, deserializing wal will use the older way, which may lead to construct a wrong block. This pr using the newest be_exec_version to deserialize wal file  to avoid compatibility problem.
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hust-hhb** [comment]
run buildall

File: be/src/vec/exec/format/wal/wal_reader.cpp
Comment (gavinchou) [inline be/src/vec/exec/format/wal/wal_reader.cpp:1]: It's better to add UT to prevent such compatibility issues.

Comment (hust-hhb) [reply be/src/vec/exec/format/wal/wal_reader.cpp:1]: now we have vwal_scanner_test ut to check this problem, and revert the wal file to old version,next time we can find out the compatibility issues by using this old wal file

**Comment by hust-hhb** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Comment by mymeiyi** [comment]
I think the problem here is that the block deserializing method should be compatible with all `be_exec_version`, rather than for version 3, use version 2 deserializing method.

File: be/src/common/config.h [hunk 1]
```diff
@@ -1198,6 +1198,7 @@ DECLARE_mInt32(group_commit_queue_mem_limit);
 // group_commit_wal_max_disk_limit=1024 or group_commit_wal_max_disk_limit=10% can be automatically identified.
 DECLARE_mString(group_commit_wal_max_disk_limit);
 DECLARE_Bool(group_commit_wait_replay_wal_finish);
+DECLARE_Int32(group_commit_wal_version);
```
Comment (mymeiyi) [inline be/src/common/config.h:1201]: should not be a config, because user may change it

**Comment by hust-hhb** [comment]
run buildall

File: be/src/common/config.h (hunk 1 above)
Comment (hust-hhb) [reply be/src/common/config.h]: already change the code,take a look please @mymeiyi

**Comment by hust-hhb** [comment]
run buildall

**Review by mymeiyi — APPROVED** [review]

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by mymeiyi** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**